examples/02_authtoken/src/App.tsx

```diff
@@ -1,29 +1,39 @@
 import React, { Suspense } from 'react';
 import { atom, useAtom } from 'jotai';
-import { createTRPCClient } from '@trpc/client';
-import { createAtomCreators } from 'jotai-trpc';
+import { createTRPCProxyClient, httpLink } from '@trpc/client';
+import { createTRPCJotai } from 'jotai-trpc';
 import { trpcPokemonUrl } from 'trpc-pokemon';
 import type { PokemonRouter } from 'trpc-pokemon';
 import { ErrorBoundary } from 'react-error-boundary';
 import type { FallbackProps } from 'react-error-boundary';
 
-const { atomWithQuery } = createAtomCreators<PokemonRouter>({
-  url: trpcPokemonUrl,
+const trpc = createTRPCJotai<PokemonRouter>({
+  links: [
+    httpLink({
+      url: trpcPokemonUrl,
+    }),
+  ],
 });
 
 const tokenAtom = atom('');
 
 const clientAtom = atom((get) =>
-  createTRPCClient<PokemonRouter>({
-    url: get(tokenAtom) ? trpcPokemonUrl : 'Invalid URL',
-    headers: {
-      Authorization: `Bearer ${get(tokenAtom)}`,
-    },
+  createTRPCProxyClient<PokemonRouter>({
+    links: [
+      httpLink({
+        url: get(tokenAtom) ? trpcPokemonUrl : 'Invalid URL',
+        headers: {
+          Authorization: `Bearer ${get(tokenAtom)}`,
+        },
+      }),
+    ],
   }),
 );
 
-const pokemonAtom = atomWithQuery('pokemon.all', [], undefined, (get) =>
-  get(clientAtom),
+const pokemonAtom = trpc.pokemon.all.atomWithQuery(
+  undefined,
+  undefined,
+  (get) => get(clientAtom).pokemon.all as any, // FIXME avoid any
```

Comment:
@sachinraja One remaining issue is this `FIXME`. Can you help??

Reply:
I'm confused, what's the purpose of `getProcedure` when we already have the path (`pokemon.all`)?

Reply:
We don't directly use React Context, so there should be an optional way to inject a custom client.

Check out v9 based API:
https://github.com/jotai-labs/jotai-trpc/blob/f56509e971db886c8c12f906e00ba792deef7879/examples/02_authtoken/src/App.tsx#L10-L27

Reply:
On second thought, it's not nice in DX. Let me see if I can pass paths around.